### PR TITLE
make.bat: optimize make.bat

### PR DESCRIPTION
**Pull request by yuyi98**
This PR optimize make.bat, fix recurring compilation errors.
- Cancel multiple self-compiling operations.
- Use `v self` instead of script Instruction.
- Clean temp files and dirs when failed.

### CODE REVIEW

File: make.bat [hunk 1]
```diff
@@ -70,22 +67,19 @@ if exist "%InstallDir%\Common7\Tools\vsdevcmd.bat" (
...
+cl.exe /nologo /w /volatile:ms /Fo%ObjFile% /O2 /MD /D_VBOOTSTRAP vc\v_win.c user32.lib kernel32.lib advapi32.lib shell32.lib /link /NOLOGO /OUT:v.exe /INCREMENTAL:NO
 if %ERRORLEVEL% NEQ 0 (
     echo cl.exe failed to build V
     goto :compileerror
 )
 
 echo rebuild from source (twice, in case of C definitions changes)
-v2.exe -o v3.exe cmd/v
-v3.exe -o v -prod cmd/v
+v self
```
Comment (alexesprit) [inline make.bat]: Now it's missing `prod` flag.

File: make.bat [hunk 2]
```diff
@@ -28,27 +27,25 @@ if not exist "%gccpath%" (
...
     exit /b 1
 )
 
 echo Now using V to build V...
-v2.exe -o v3.exe cmd/v
-v3.exe -o v.exe -prod cmd/v
+v self
 if %ERRORLEVEL% NEQ 0 (
     echo v.exe failed to compile itself - Create an issue at 'https://github.com/vlang'
+	rd /s /q vc
```
Comment (alexesprit) [inline make.bat]: Indentation issue is here.

File: make.bat [hunk 3]
```diff
@@ -28,27 +27,25 @@ if not exist "%gccpath%" (
     goto:msvcstrap
 )
 
-gcc -std=c99 -municode -w -o v2.exe vc\v_win.c
+gcc -std=c99 -municode -w -o v.exe vc\v_win.c
 if %ERRORLEVEL% NEQ 0 (
     echo gcc failed to compile - Create an issue at 'https://github.com/vlang'
```
Comment (alexesprit) [inline make.bat]: I would also unify error output. We can goto `:compileerror` here instead of exiting with 1.

File: make.bat (hunk 1 above)
Comment (yuyi98) [reply make.bat]: OK, you're right!

File: make.bat (hunk 3 above)
Comment (yuyi98) [reply make.bat]: Because always run make.bat compilation error, only to modify it, I hope there is a better way.

**Pull request closed without merging**